src/hb-ot-cmap-table.hh

```diff
@@ -1468,7 +1503,155 @@ struct cmap
     | hb_filter ([&] (const hb_pair_t<hb_codepoint_t, hb_codepoint_t> _)
 		 { return (_.second != HB_MAP_VALUE_INVALID); })
     ;
-    cmap_prime->serialize (c->serializer, it, encodingrec_iter, this, c->plan);
+
+    for (auto _ : format12_tables.iter ())
```

Comment:
Is it safe to iterate a map that's being modified?

Reply:
Looking at the map impl, I don't think it is. It's possible for the map's resize() to be triggered by a del() call since that internally calls set_with_hash(). Specifically if the last addition to the map pushes it past the occupancy limit for a resize, then the next del() call will trigger a resize() and the array which the iterator is iterating through will be deleted.

Reply:
So you'll instead need to create keep a set/list of the format12_tables to be deleted and add to it in this loop.

Reply:
In set_with_hash (), resize () won't be triggered if the value is kINVALID. 
{
if (unlikely (key == kINVALID)) return;
if ((occupancy + occupancy / 2) >= mask && !resize ()) return;
}
It seems safe to call del () while iterating a map.

Reply:
oh, sorry, the value is invalid not the key. I will change it.